skyfield/data/mpc.py

```diff
@@ -226,6 +226,30 @@ def comet_orbit(row, ts, gm_km3_s2):
     comet._rotation = inertial_frames['ECLIPJ2000'].T
     return comet
 
+def _comet_orbits(rows, ts, gm_km3_s2):
+    e = rows.eccentricity.values
+    p = 1 - e*e
+    parabolic = (e == 1.0)
+    p[parabolic] = rows.perihelion_distance_au.values[parabolic] * 2.0
+    p[~parabolic] *= rows.perihelion_distance_au.values[~parabolic]/ (1.0 - e[~parabolic])
```

Comment:
One final idea, because it looks so painful to be creating those specially indexed copies of the Pandas columns:
```
parabolic = (e == 1.0)
p = (1 - e*e) / (1.0 - e + parabolic)
p[parabolic] += 2.0
p *= rows.perihelion_distance_au.values
```
Would that produce the same output, but without needing the special indexing?

Reply:
Nice! That's a clever way to avoid the division by 0. I added that in.